### DISCUSSION TITLE
React Native Window focus not working

### DISCUSSION THREAD

**Discussion by hasanagh** [TanStack/query#2284] [Q&A]
Hello, 

I am  expecting a refetch when the app is back in focus. after I added this https://react-query.tanstack.com/guides/window-focus-refetching#managing-focus-in-react-native but I am not getting a refetch

I added this to my App.js to subscribe to AppState changes

```
focusManager.setEventListener(handleFocus => {
  AppState.addEventListener('change', handleFocus)

  return () => {
    AppState.removeEventListener('change', handleFocus)
  };
});

```

What might I be doing wrong?

**Comment by dan-lee** [comment]
I guess the example is not completely correct. You need to check for the app state to be active. That's what I am doing:
```tsx
focusManager.setEventListener(() => {
  const handleAppStateChange = (appState: AppStateStatus) => {
    focusManager.setFocused(appState === 'active')
  }

  AppState.addEventListener('change', handleAppStateChange)

  return () => {
    AppState.removeEventListener('change', handleAppStateChange)
  }
})
```

**Reply by hasanagh** [reply]
@dan-lee  in what file are you putting this code snippet? I also tried the same code and its not working.

**Reply by dan-lee** [reply]
Very top level, like `index.ts` or `App.ts`. Did you try logging something in the state change handler, if it really registers app state changes?

Looks like a similar approach is suggested here: https://github.com/tannerlinsley/react-query/pull/2152
